library/item/item_setting/src/main/java/org/quiltmc/qsl/itemsetting/api/CustomItemSetting.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.itemsetting.api;
+
+import java.util.function.Supplier;
+
+import net.minecraft.item.Item;
+
+import org.jetbrains.annotations.ApiStatus;
+import org.quiltmc.qsl.itemsetting.impl.CustomItemSettingImpl;
+
+/**
+ * An interface describing the behavior for custom item settings.
+ * This feature can be used by mods to add non-standard settings
+ * to items in a way that is compatible with other mods that add
+ * settings to items.
+ *
+ * <p>Values of this setting can be retrieved from an item using {@link CustomItemSetting#get(Item)},
+ * and users that wish to expose a custom setting for use in other mods should do so by exposing
+ * the CustomItemSetting instance.
+ *
+ * <h2>Usage Example</h2>
+ * <pre>{@code
+ * // Create the setting instance. You can think of this as the "setting key".
+ * public static final CustomItemSetting<String> CUSTOM_TOOLTIP = CustomItemSetting.create(() -> null);
+ *
+ * @Override
+ * public void onInitializeClient() {
```

Comment:
```suggestion
 * public void onInitializeClient(ModContainer mod) {
```

QSL entrypoint added a new parameter.